src/storage-client/src/controller/persist_handles.rs

```diff
@@ -48,6 +52,18 @@ enum PersistReadWorkerCmd<T: Timestamp + Lattice + Codec64> {
     Register(GlobalId, SinceHandle<SourceData, (), T, Diff, PersistEpoch>),
     Update(GlobalId, SinceHandle<SourceData, (), T, Diff, PersistEpoch>),
     Downgrade(BTreeMap<GlobalId, Antichain<T>>),
+    SnapshotStats(GlobalId, Antichain<T>, oneshot::Sender<SnapshotStatsRes<T>>),
+}
+
+/// A newtype wrapper to hang a Debug impl off of.
+pub(crate) struct SnapshotStatsRes<T>(
+    pub(crate) Box<dyn Future<Output = Result<SnapshotStats<T>, StorageError>> + Send + 'static>,
```

Comment:
I would use `Pin<Box<...>>` here or better yet `futures::future::BoxFuture<..., 'static>`, which makes it so you don't need to use `Box::into_pin`